### PR TITLE
Add the possibility to template playbooks

### PR DESCRIPTION
**Pull request by dannymc129**
I was recently recommended by @lritter to consider using this tool for some adhoc queries we need to run regularly and I ended up really liking it. Simple, clean, functional, we're already snowplow users and a perfect match.

But I wasn't happy with the idea of all of my hosts and password being hosted along with the playbooks, so to fix this I changed it so that the `-var` flag will pass into a playbook similar to the way it does for sql templates, then we can get our passwords out of local config without worrying the tool. 

I also found out that based on:
https://github.com/snowplow/sql-runner/blob/master/sql_runner/options.go#L28-L36

there's a limit of only a single key=value pair in the `-var` flag, I extended this so that it will load any number of vars via comma separation (`key=value,key2=value2`)

Pretty new to go, please do pick it apart.

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

📝 Please visit https://github.com/snowplow/snowplow/wiki/CLA to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

File: sql_runner/consul_provider.go [hunk 1]
```diff
@@ -15,12 +15,14 @@ package main
 type ConsulPlaybookProvider struct {
 	consulAddress string
 	consulKey     string
+	variables 		CLIVariables
```
Comment (BenFradet) [inline sql_runner/consul_provider.go]: the indent seems off

File: sql_runner/consul_provider.go [hunk 2]
```diff
@@ -15,12 +15,14 @@ package main
...
 }
 
-func NewConsulPlaybookProvider(consulAddress, consulKey string) *ConsulPlaybookProvider {
+func NewConsulPlaybookProvider(options Options) *ConsulPlaybookProvider {
 	return &ConsulPlaybookProvider{
-		consulAddress: consulAddress,
-		consulKey:     consulKey,
+		consulAddress: options.consul,
+		consulKey:     options.playbook,
+		variables: 		 options.variables,
```
Comment (BenFradet) [inline sql_runner/consul_provider.go]: same, don't hesitate to run `gofmt`

File: sql_runner/consul_provider.go [hunk 3]
```diff
@@ -15,12 +15,14 @@ package main
 type ConsulPlaybookProvider struct {
 	consulAddress string
 	consulKey     string
+	variables 		CLIVariables
 }
 
-func NewConsulPlaybookProvider(consulAddress, consulKey string) *ConsulPlaybookProvider {
+func NewConsulPlaybookProvider(options Options) *ConsulPlaybookProvider {
```
Comment (BenFradet) [inline sql_runner/consul_provider.go]: are we sure we want to pass the whole options bundle and not just the address, key and variables?

File: sql_runner/yaml_provider.go [hunk 4]
```diff
@@ -14,11 +14,13 @@ package main
 
 type YAMLFilePlaybookProvider struct {
 	playbookPath string
+	variables CLIVariables
 }
 
-func NewYAMLFilePlaybookProvider(playbookPath string) *YAMLFilePlaybookProvider {
+func NewYAMLFilePlaybookProvider(options Options) *YAMLFilePlaybookProvider {
```
Comment (BenFradet) [inline sql_runner/yaml_provider.go]: same remark as above

File: sql_runner/yaml_utils.go [hunk 5]
```diff
@@ -26,13 +27,18 @@ var (
 
 // Parses a playbook.yml to return the targets
 // to execute against and the steps to execute
-func parsePlaybookYaml(playbookBytes []byte) (Playbook, error) {
+func parsePlaybookYaml(playbookBytes []byte, variables CLIVariables) (Playbook, error) {
```
Comment (BenFradet) [inline sql_runner/yaml_utils.go]: might be worth it to restrict ourselves to just a `map[string]string`

File: sql_runner/yaml_utils.go [hunk 6]
```diff
@@ -51,3 +57,17 @@ func cleanYaml(rawYaml []byte) []byte {
...
+	if err != nil {
+		return "", err
+	}
+
+	var filled bytes.Buffer
+	if err := t.Execute(&filled, variables); err != nil {
+		return "", err
+	}
+
+	return string(filled.String()), err
```
Comment (BenFradet) [inline sql_runner/yaml_utils.go]: why the wrapping of a string in another string?

File: sql_runner/yaml_utils.go [hunk 7]
```diff
@@ -51,3 +57,17 @@ func cleanYaml(rawYaml []byte) []byte {
 	}
 	return buffer.Bytes()
 }
+
+func fillPlaybookTemplate(playbookStr string, variables CLIVariables) (string, error) {
```
Comment (BenFradet) [inline sql_runner/yaml_utils.go]: same I think we can drop `CLIVariables` in favour of `map[string]string`

File: sql_runner/options.go [hunk 8]
```diff
@@ -26,11 +26,12 @@ func (i *CLIVariables) String() string {
...
-	if len(split) > 1 {
-		key := split[0]
-		val := split[1]
-		(*i)[key] = val
+	var split = strings.Split(value, ",")
+
+	for value := range split {
+		kv := strings.SplitN(split[value], "=", 2)
+
+		(*i)[kv[0]] = kv[1]
```
Comment (BenFradet) [inline sql_runner/options.go]: can we check the length of kv and send an error accordingly to avoid index out of range?

**Comment by dannymc129** [comment]
I signed it!

**Comment by snowplowcla** [comment]
Confirmed! @dannymc129 has signed the Individual Contributor License Agreement. Thanks so much

**Comment by BenFradet** [comment]
Merging, sorry for the delay 👍

**Comment by alexanderdean** [comment]
@BenFradet I think this should have been split into two commits - one to add the new templating functionality, and the other to fix the limitation around the single variable provision.

**Comment by BenFradet** [comment]
Nw, we can still split the commit when merging to master.

I'll log the other issue.

**Comment by alexanderdean** [comment]
Thanks @BenFradet !

**Pull request merged**